### PR TITLE
add vsag library

### PR DESCRIPTION
**Pull request by wxyucs**
add vsag library, and also fixes: #547

### CODE REVIEW

File: ann_benchmarks/main.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def run_worker(cpu: int, args: argparse.Namespace, queue: multiprocessing.Queue)
             run(definition, args.dataset, args.count, args.runs, args.batch)
         else:
             memory_margin = 500e6  # reserve some extra memory for misc stuff
-            mem_limit = int((psutil.virtual_memory().available - memory_margin) / args.parallelism)
+            mem_limit = int((psutil.virtual_memory().total - memory_margin) / args.parallelism)
```
Comment (maumueller) [inline ann_benchmarks/main.py:73]: Please remove this

**Comment by maumueller** [comment]
Thanks for the contribution @wxyucs. Please also add your implementation [to the CI](https://github.com/erikbern/ann-benchmarks/blob/main/.github/workflows/benchmarks.yml).

File: ann_benchmarks/main.py (hunk 1 above)
Comment (wxyucs) [reply ann_benchmarks/main.py]: Hi @maumueller, this change is related to #547. And now we just modified it with the suggested way in the issue. Please have a check.

**Comment by wxyucs** [comment]
> Thanks for the contribution @wxyucs. Please also add your implementation [to the CI](https://github.com/erikbern/ann-benchmarks/blob/main/.github/workflows/benchmarks.yml).

@maumueller Already done, please take a look.

File: ann_benchmarks/main.py (hunk 1 above)
Comment (maumueller) [reply ann_benchmarks/main.py]: Thanks for modifying it in the suggested way.

**Comment by maumueller** [comment]
Thanks for the updates. Could you confirm that the CI is failing for your implementation due to the necessary SIMD/AVX operations not being present on the CI server?

**Comment by erikbern** [comment]
Looks like it's passing now – great

**Comment by wxyucs** [comment]
Hi @maumueller, we fixed the issue that pyvsag cannot run on some platforms without SIMD/AVX instructions. And now the CI test for pyvsag is passed.

**Comment by maumueller** [comment]
Thanks for the contribution!

**Comment by wxyucs** [comment]
Hi @maumueller, I saw that this pull request has been merged. Is that any plan for running the new algorithm?

**Pull request merged**